pyneuroml/swc/ExportNML.py

```diff
@@ -0,0 +1,550 @@
+            :type segment_type: int
+            :return: List of group names the segment belongs to.
+            :rtype: List[str]
+            """
+            groups = ["all"]
+            if segment_type == SWCNode.SOMA:
+                groups.extend(["soma_group", "color_white"])
+            elif segment_type == SWCNode.AXON:
+                groups.extend(["axon_group", "color_grey"])
+            elif segment_type == SWCNode.BASAL_DENDRITE:
+                groups.extend(["basal_dendrite", "dendrite_group", "color_green"])
+            elif segment_type == SWCNode.APICAL_DENDRITE:
+                groups.extend(["apical_dendrite", "dendrite_group", "color_magenta"])
+            elif segment_type >= 5:
+                groups.extend([f"SWC_group_{segment_type}", "dendrite_group"])
+            elif segment_type == 0:
+                groups.extend(["SWC_group_0_assuming_soma", "soma_group"])
+            elif segment_type == -1:
+                groups.extend(["SWC_group_-1_assuming_soma", "soma_group"])
+            return groups
+
+        # Initialize all groups
+        all_groups = set(
+            [
+                "all",
+                "soma_group",
+                "axon_group",
+                "dendrite_group",
+                "basal_dendrite",
+                "apical_dendrite",
+            ]
+        )
+        all_groups.update([f"SWC_group_{i}" for i in range(-1, 10)])
+        all_groups.update(["color_white", "color_grey", "color_green", "color_magenta"])
+
+        group_members: Dict[str, Set[int]] = {group: set() for group in all_groups}
+
+        # Assign segments to groups
+        for segment in self.cell.morphology.segments:
+            segment_type = segment.type
+            groups = get_groups_for_type(segment_type)
+
+            for group in groups:
+                group_members[group].add(segment.id)
+
+        # Create SegmentGroup objects
+        for group_name, members in group_members.items():
+            if members:
+                group = SegmentGroup(id=group_name)
+                for member_id in sorted(members):  # Sort to ensure consistent order
+                    group.members.append(Member(segments=member_id))
+                self.cell.morphology.segment_groups.append(group)
+
+        # Add cable groups (assuming we're using NeuroML v2)
+        for cable_id, segments in self.cable_ids_vs_indices.items():
+            cable_group = SegmentGroup(id=f"{self.cable_prefix_v2}{cable_id}")
+            for seg_id in sorted(set(segments)):  # Remove duplicates and sort
+                cable_group.members.append(Member(segments=seg_id))
+            self.cell.morphology.segment_groups.append(cable_group)
+
+        # Add the cell_type property
+        self.cell.properties.append(
+            Property(tag="cell_type", value="converted_from_swc")
+        )
+
+        logger.info("Segment groups created successfully")
+
+    def print_soma_segments(self) -> None:
```

Comment:
is this for debugging?

you can use `cell.get_segment_group("soma_group")` to get all segments in the segment group.

See `help(neuroml.Cell)` in a python interpreter to see all the methods of the `Cell` class.

Reply:
yes I added this for debugging...will go through the recomendation

Reply:
Cool, I'd also suggest writing unit tests for testing that each part of your code is doing what you expect it to do :)